### PR TITLE
topology: sof-tgl-rt711-rt1308: align DMIC support with HDaudio

### PR DESCRIPTION
**Pull request by plbossart**
Somehow we use different configurations for the DMIC, remove
everything in this file and rely on the same macros.

The only difference with HDaudio is the pipeline and PCM device numbers

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Review by singalsu — APPROVED** [review]

File: tools/topology/sof-tgl-rt711-rt1308.m4 [hunk 1]
```diff
@@ -56,20 +74,6 @@ PIPELINE_PCM_ADD(sof/pipe-volume-capture.m4,
 	1000, 0, 0,
 	48000, 48000, 48000)
 
-# Low Latency playback pipeline 3 on PCM 2 using max 2 channels of s32le.
-# Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
-	3, 2, 2, s32le,
-	1000, 0, 0,
-	48000, 48000, 48000)
```
Comment (ranj063) [inline tools/topology/sof-tgl-rt711-rt1308.m4:64]: was this intentional @plbossart or did you mean to remove pipeline 5?

Comment (plbossart) [reply tools/topology/sof-tgl-rt711-rt1308.m4:64]: nope, that's probably the issue. thanks!

**Comment by plbossart** [comment]
@ranj063 I just want to see if the topology is correctly loaded with the dependencies on macros. 

two opens on my side:
1. What I don't recall is how these macro work in practice since we have different clocks between CML and ICL/TGL.

2. we may have to also fix the PCM device number. It's fixed in UCM, but it varies across different SoundWire topologies depending on the platform. We may have to align with Chrome and use invariant device number 99 and 100 to be 'bug compatible'. that was flagged by @libinyang and I think he's right.

**Comment by ranj063** [comment]
> * What I don't recall is how these macro work in practice since we have different clocks between CML and ICL/TGL.
> *

Sorry, Pierre, Im a bit confused. Which macros and clocks are we talking about here?

**Comment by plbossart** [comment]
SOFCI TEST

**Review by juimonen — APPROVED** [review]
dmic volume names look now good the me, so approved

**Comment by plbossart** [comment]
> > * What I don't recall is how these macro work in practice since we have different clocks between CML and ICL/TGL.
> 
> Sorry, Pierre, Im a bit confused. Which macros and clocks are we talking about here?

maybe I am confused in the first place why the existing cml-rt700.m4 and icl-rt700.m4 topologies use the same DMIC config. I thought we needed to use different configs depending on the xtal used.

````
# dmic01 (ID: 1)
DAI_CONFIG(DMIC, 0, 2, dmic01,
	   DMIC_CONFIG(1, 500000, 4800000, 40, 60, 48000,
		DMIC_WORD_LENGTH(s32le), 400, DMIC, 0,
		PDM_CONFIG(DMIC, 0, FOUR_CH_PDM0_PDM1)))

# dmic16k (ID: 2)
DAI_CONFIG(DMIC, 1, 3, dmic16k,
	   DMIC_CONFIG(1, 500000, 4800000, 40, 60, 16000,
		DMIC_WORD_LENGTH(s16le), 400, DMIC, 1,
		PDM_CONFIG(DMIC, 1, STEREO_PDM0)))
````

we use the same configs here but the xtals are different (24 MHz and 38.4 respectively). @singalsu was the idea that we use 4.8MHz as a common clock setup and use a division by 5 and 8 in the firmware?

**Review by lgirdwood — APPROVED** [review]
Lets get this in for rc1 and we can make any adjustment prior to v1.7 final.

**Comment by singalsu** [comment]
@plbossart In the above DMIC_CONFIG() macros the 500 kHz is set as min. clock and 4.8 MHz as max. clock. There's no common rates targeted though 2.4 MHz mic clock happens often. The driver will determine the suitable PDM clock divisors to reach a suitable oversampling for the requested sample rate. Min. oversampling is not exposed to topology, it's hard-coded to at least 50 in firmware. Lower oversampling ratios towards 50 are preferred to save power, though some FIR decimation factors compatibility for FIFOs A & B rates also impact the selection.

**Pull request merged**